packages/@sanity/components/src/edititem/PopOver.js

```diff
@@ -21,248 +25,269 @@ export default class EditItemPopOver extends React.Component {
       handleClick: PropTypes.func
     })),
     fullWidth: PropTypes.bool,
-    onNeedScroll: PropTypes.func,
+    isOpen: PropTypes.bool,
     scrollContainer: PropTypes.node,
-    scrollContainerId: PropTypes.string,
-    isOpen: PropTypes.bool
-  }
+    scrollContainerId: PropTypes.string
+  };
 
   static defaultProps = {
     onClose() {}, // eslint-disable-line
-    scrollContainerId: 'Sanity_Default_DeskTool_Editor_ScrollContainer',
     actions: [],
-    isOpen: true
+    isOpen: true,
+    scrollContainerId: 'Sanity_Default_DeskTool_Editor_ScrollContainer'
+  };
+
+  _rootElement = null
+  _portalModalElement = null
+  _arrowElement = null
+  _scrollContainerElement = null
+  _initialScrollTop = null
+  _scrollContainerElementInitialPaddingBottom = null
+  _translateYHistory = []
+  _initialBodyStyleOverflow = null
+
+  _scrollOptions = {
+    duration: 250,
+    ease: ease.easeInOutQuart
   }
 
-  constructor(props) {
-    super()
-    this.handleResize = debounce(this.handleResize.bind(this), 17) // 60fps
-    this.scrollOptions = {
-      duration: 250,
-      ease: ease.easeInOutQuart
-    }
-    this.state = {
-      scrollContainer: null,
-      rootOffsetTop: 0,
-      modalTranslateY: 0,
-      fullHeight: false,
-      portalModalRects: {
-        top: undefined,
-        left: undefined,
-        right: undefined,
-        height: undefined,
-        bottom: undefined
-      }
-    }
+  _elementResizeDetector = null
+
+  _portalModalRects = {
+    top: undefined,
+    left: undefined,
+    right: undefined,
+    height: undefined,
+    bottom: undefined
   }
 
-  handleClose = () => {
-    const {scrollContainer} = this.state
-    if (scrollContainer) {
-      scroll.top(scrollContainer, this.initialScrollTop, this.scrollOptions, () => {
-        scrollContainer.style.paddingBottom = '0'
-      })
-    }
-    this.props.onClose()
+  state = {
+    portalIsOpen: false,
+    isFocused: false
   }
 
-  repositionElement() {
-    const {scrollContainer} = this.state
-    const {rootRects} = this.state
-    const portalModalElement = this._portalModalElement
+  componentDidMount() {
+    popOverStack.push(this)
 
-    if (!portalModalElement || !rootRects) {
-      return
+    if (popOverStack[0] === this) {
+      const {scrollContainerId, scrollContainer} = this.props
+      if (scrollContainerId) {
+        this._scrollContainerElement = document.getElementById(scrollContainerId)
```

Comment:
What if `document.getElementById(scrollContainerId)` returns `null`? Should it be an error?